lib/OpenQA/JobSettings.pm

```diff
@@ -17,6 +17,35 @@ package OpenQA::JobSettings;
 use strict;
 use warnings;
 
+sub generate_settings {
+    my ($params) = @_;
+    my $settings = $params->{settings};
+    my @worker_class;
+    for my $entity (qw (product machine test_suite job_template)) {
+        next unless $params->{$entity};
+        my @entity_settings = $params->{$entity}->settings;
+        for my $setting (@entity_settings) {
+            if ($setting->key eq 'WORKER_CLASS') {
+                push @worker_class, $setting->value;
+                next;
+            }
+            $settings->{$setting->key} = $setting->value;
```

Comment:
how about

```perl
$settings->{$setting->key} = $setting->value unless $setting->key eq 'WORKER_CLASS';
push @worker_class, $setting->value;
```

Reply:
we also need a `next` in unless. However, I do not see a difference between using `if` or `unless`. If you do not mind, I prefer to keep that.

Reply:
The version from @okurz is not equivalent, indeed. I'd keep it as well.